### PR TITLE
remember last workspace visited by user

### PR DESCRIPTION
**Pull request by vieiralucas**

### CODE REVIEW

File: apps/web/src/hooks/useSessionRedirect.ts [hunk 1]
```diff
@@ -38,10 +38,19 @@ export const useSessionRedirect = (shouldRedirect = true) => {
...
-      router.replace(`/workspaces/${workspaces.data[0].id}/documents`)
+      const user = session.data
+      const workspace =
+        workspaces.data.find(
+          (workspace) => workspace.id === user.lastVisitedWorkspaceId
+        ) ??
+        workspaces.data.find((workspace) => workspace.ownerId === user.id) ??
+        workspaces.data[0]
+      if (workspace) {
+        router.replace(`/workspaces/${workspaces.data[0].id}/documents`)
```
Comment (lucasfcosta) [inline apps/web/src/hooks/useSessionRedirect.ts:50]: Can't `workspaces.data[0]` be undefined, technically speaking? Shouldn't we have kept that `length` check above?

File: apps/web/src/pages/workspaces/index.tsx [hunk 2]
```diff
@@ -7,13 +7,19 @@ export default function WorkspacesPage() {
...
       return
     }
 
-    const workspace = data[0]
+    const workspace =
+      data.find(
+        (workspace) => workspace.id === session.data?.lastVisitedWorkspaceId
+      ) ??
+      data.find((workspace) => workspace.ownerId === session.data?.id) ??
+      data[0]
```
Comment (lucasfcosta) [inline apps/web/src/pages/workspaces/index.tsx:22]: Probably not something for this PR, but it's kind of annoying that we have to have very similar code here and in the session redirect.

File: packages/database/prisma/migrations/20241129202947_add_last_visited_workspace_id/migration.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-- AlterTable
+ALTER TABLE "User" ADD COLUMN     "lastVisitedWorkspaceId" UUID;
+
+-- AddForeignKey
+ALTER TABLE "User" ADD CONSTRAINT "User_lastVisitedWorkspaceId_fkey" FOREIGN KEY ("lastVisitedWorkspaceId") REFERENCES "Workspace"("id") ON DELETE SET NULL ON UPDATE CASCADE;
```
Comment (lucasfcosta) [inline packages/database/prisma/migrations/20241129202947_add_last_visited_workspace_id/migration.sql:5]: Can we make this migration set the first value of `lastVisitedWorkspaceId` as the _latest_ workspace this user created? If that doesn't exist, we could just set it to the _earliest_ workspace.

Comment (vieiralucas) [reply packages/database/prisma/migrations/20241129202947_add_last_visited_workspace_id/migration.sql:5]: The source code will default to one he is owner of

File: apps/web/src/hooks/useSessionRedirect.ts (hunk 1 above)
Comment (vieiralucas) [reply apps/web/src/hooks/useSessionRedirect.ts:50]: This is a bug, it should use `workspace.id`

**Pull request merged**